src/OrleansRuntime/Placement/IPlacementContext.cs

```diff
@@ -54,22 +56,32 @@ internal interface IPlacementContext
         /// <returns></returns>
         bool TryGetActivationData(ActivationId id, out ActivationData activationData);
 
-        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, string genericArguments = null);
+        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, out MultiClusterRegistrationStrategy strategy, string genericArguments = null);
     }
 
     internal static class PlacementContextExtensions
     {
-        public static Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
+        public static async Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
         {
             List<ActivationAddress> l;
-            return @this.FastLookup(grainId, out l) ? Task.FromResult(l) : @this.FullLookup(grainId);
+            if (@this.FastLookup(grainId, out l))
+            {
+                return l;
+            }
+            else
+            {
+                var result = await @this.FullLookup(grainId);
```

Comment:
`FullLookup` needs to do a remote directory lookup, its a grain call under the covers. So it can't return a prompt answer.

There is a lot of things we can micro optimize here. Like in most cases we don't need a List, but a single item. Or break the whole lookup call into 2 calls - prompt and async. I would leave the extra optimizations for a separate un-related PR but make sure that what ever we are changing here in this PR does not induce an extra overhead.